src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs

```diff
@@ -184,13 +189,13 @@ private ISocketWrapper GetConnection()
             ISocketWrapper socket = null;
             try
             {
-                Thread thread = Thread.CurrentThread;
+                Thread thread = _isRunningRepl ? null : Thread.CurrentThread;
                 MemoryStream payloadMemoryStream = s_payloadMemoryStream ??= new MemoryStream();
                 payloadMemoryStream.Position = 0;
                 PayloadHelper.BuildPayload(
                     payloadMemoryStream,
                     isStatic,
-                    thread.ManagedThreadId,
+                    thread == null ? 1 : thread.ManagedThreadId,
```

Comment:
I think we should just not support multiple threads in repl scenarios.

Reply:
Although I think we may be in trouble if we're using `ForeachBatch` in the repl session.

Reply:
So, `ForeachBatch` does not work in repl now with this change?

Reply:
If on the JVM side, everything will be scheduled on one ExecutorService, then I don't think it'll work.  Let's say you set up the `StreamingQuery` and you call either `AwaitTermination` or `ProcessAllAvailable`.  This will be scheduled on the `ExecutorService` and will be blocking.  Then a spark thread will spin up and call the `ForeachBatch` callback.  On the dotnet side, the `ForeachBatch` callback will run on a new thread, but when the dataframe methods are called, it will be scheduled on the same `ExecutorService`  that is currently blocked by `AwaitTermination`/`ProcessAllAvailable` ?